src/analyses/constant_propagator.cpp

```diff
@@ -278,30 +278,53 @@ bool constant_propagator_domaint::ai_simplify(
 
 bool constant_propagator_domaint::valuest::is_constant(const exprt &expr) const
 {
-  if(expr.id()==ID_side_effect &&
-     to_side_effect_expr(expr).get_statement()==ID_nondet)
-    return false;
+  if(expr.is_constant())
+    return true;
 
```

Comment:
Is there any way we could share the code between this and symex rather than keeping the two maintained in parallel? expr_util.h?

Reply:
I'll give this a try, even though symex has some peculiarities around plus and multiplication just so as to feed more information to calls to `malloc`.